packages/cloudflare/test/handler.test.ts

```diff
@@ -0,0 +1,300 @@
+    vi.clearAllMocks();
+  });
+
+  test('gets env from handler', async () => {
+    const handler = {
+      fetch(_request, _env, _context) {
+        return new Response('test');
+      },
+    } satisfies ExportedHandler;
+
+    const optionsCallback = vi.fn().mockReturnValue({});
+
+    const wrappedHandler = withSentry(optionsCallback, handler);
+    await wrappedHandler.fetch(new Request('https://example.com'), MOCK_ENV, createMockExecutionContext());
+
+    expect(optionsCallback).toHaveBeenCalledTimes(1);
+    expect(optionsCallback).toHaveBeenLastCalledWith(MOCK_ENV);
+  });
+
+  test('passes through the response from the handler', async () => {
+    const response = new Response('test');
+    const handler = {
+      async fetch(_request, _env, _context) {
+        return response;
+      },
+    } satisfies ExportedHandler;
+
+    const wrappedHandler = withSentry(() => ({}), handler);
+    const result = await wrappedHandler.fetch(
+      new Request('https://example.com'),
+      MOCK_ENV,
+      createMockExecutionContext(),
+    );
+
+    expect(result).toBe(response);
+  });
+
+  test('flushes the event after the handler is done using the cloudflare context.waitUntil', async () => {
+    const handler = {
+      async fetch(_request, _env, _context) {
+        return new Response('test');
+      },
+    } satisfies ExportedHandler;
+
+    const context = createMockExecutionContext();
+    const wrappedHandler = withSentry(() => ({}), handler);
+    await wrappedHandler.fetch(new Request('https://example.com'), MOCK_ENV, context);
+
+    // eslint-disable-next-line @typescript-eslint/unbound-method
+    expect(context.waitUntil).toHaveBeenCalledTimes(1);
+    // eslint-disable-next-line @typescript-eslint/unbound-method
+    expect(context.waitUntil).toHaveBeenLastCalledWith(expect.any(Promise));
+  });
+
+  test('creates a cloudflare client and sets it on the handler', async () => {
+    const handler = {
+      async fetch(_request, _env, _context) {
+        expect(SentryCore.getClient() instanceof CloudflareClient).toBe(true);
+        return new Response('test');
+      },
+    } satisfies ExportedHandler;
+
+    const context = createMockExecutionContext();
+    const wrappedHandler = withSentry(() => ({}), handler);
+    await wrappedHandler.fetch(new Request('https://example.com'), MOCK_ENV, context);
+
+    expect.assertions(1);
+  });
+
+  describe('scope instrumentation', () => {
+    test('adds cloud resource context', async () => {
+      const handler = {
+        async fetch(_request, _env, _context) {
+          SentryCore.captureMessage('test');
+          return new Response('test');
+        },
+      } satisfies ExportedHandler;
+
+      let sentryEvent: Event = {};
+      const wrappedHandler = withSentry(
+        (env: any) => ({
+          dsn: env.MOCK_DSN,
+          beforeSend(event) {
+            sentryEvent = event;
+            return null;
+          },
+        }),
+        handler,
+      );
```

Comment:
l: can we extract this into a helper just call that in the next few tests?

Reply:
I'll come back to take a look at this after this PR merges, good suggestion.